src/app/features/library/search-form/search-form.component.ts

```diff
@@ -0,0 +1,118 @@
+import { Component, Inject, OnInit } from '@angular/core';
+
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { HelperService } from '../../../core/services/utils/helper.service';
+import { VIMEO_VIDEO_TYPE, YT_VIDEO_TYPE } from '../../../core/config/videos-type.config';
+import { YtDataService } from '../../../core/services/data-integration/yt-data.service';
+import { VimeoDataService } from '../../../core/services/data-integration/vimeo-data.service';
+import { YtVideosResponse } from '../../../shared/interfaces/youtube/yt-videos-response.interface';
+import { VimeoResponse } from '../../../shared/interfaces/vimeo/vimeo-response.interface';
+import { SavedVideoData } from '../../../shared/interfaces/saved-video-data.interface';
+import { VideosStoreService } from '../../../core/services/data-integration/videos-store.service';
+import { SnackbarService } from '../../../core/services/utils/snackbar.service';
+import { catchError } from 'rxjs/operators';
+import { HttpErrorResponse } from '@angular/common/http';
+import { of } from 'rxjs';
+
+@Component({
+  selector: 'app-search-form',
+  templateUrl: './search-form.component.html',
+  styleUrls: ['./search-form.component.scss'],
+  providers: [
+    { provide: YT_VIDEO_TYPE, useValue: YT_VIDEO_TYPE },
+    { provide: VIMEO_VIDEO_TYPE, useValue: VIMEO_VIDEO_TYPE },
+  ],
+})
+export class SearchFormComponent implements OnInit {
+
+  public form: FormGroup;
+  private readonly VIDEO_NOT_FOUND_MESSAGE = 'Video not found';
+
+  constructor(
+    private fb: FormBuilder,
+    private helperService: HelperService,
+    private ytDataService: YtDataService,
+    private vimeoDataService: VimeoDataService,
+    private videosStoreService: VideosStoreService,
+    private snackbarService: SnackbarService,
+    @Inject(YT_VIDEO_TYPE) public ytVideoType: string,
+    @Inject(VIMEO_VIDEO_TYPE) public vimeoVideoType: string,
+  ) {}
+
+  public ngOnInit(): void {
+    this.form = this.createForm();
+  }
+
+  private clearInput(): void {
+    this.form.controls.id.reset();
+  }
+
+  public getDemoVideos(): void {
+    this.videosStoreService.getDemoVideos();
+  }
+
+  public onSubmit(): void {
+    const id = this.helperService.extractId(this.form.value.id);
+    const type = this.form.value.type;
+    let foundVideo: SavedVideoData;
+
+    if (id.trim() === '') {
+      return;
+    }
+
+    if (type === YT_VIDEO_TYPE) {
```

Comment:
Nawiązując do tego komentarza wyżej, takie coś nie przejdzie cwaniaczku :P Nie ma być wyboru filmu jaki szukasz, sam masz to ogarnąć w serwisach. W treści zadania było, że ma być pisana apka z myślą, że takich serwisów jak YT czy Vimeo może by w przyszłości więcej. Wyobrażasz sobie wtedy ten kod? Jeszcze więcej ifów i kodu.

Reply:
Oh, no dobła no :feels:

Reply:
Winny się tłumaczy. Wtedy lepiej dać dialog "Który film chcesz dodać".

Reply:
dobra, dobra. Wtedy to już by nie była taka prosta apka :trollface: